lib/controllers/project.js

```diff
@@ -588,6 +588,48 @@ Project.prototype.scale = function (projectId, instances, await, callback) {
   });
 };
 
+/**
+ * Downloads the source code as a tarball for the specified project id.
+ * @param {string} projectId - the ID of the project.
+ * @param {string} projectName - the name of the project
+ * @param {string} outLocation - the path to the directory where the tarball should be downloaded
+ * @param {function} callback - function invoked with the file stream. Responsible for the actual download.
+ */
+//-----------------------------------------------------------------------------
+Project.prototype.download = function(projectId, projectName, outLocation, callback) {
+  librarian.project.download(projectId, userConfig.data.apiKey, function (err, req, res) {
+    if (err) return callback(Errors.getMessage(err));
+
+
+    var date = new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '');
```

Comment:
Slightly simpler format string that we already use in `emmet`:

`new Date().toISOString().replace(/:/g, '.') // 2016-01-15T20.39.14.868Z`